hedera-node/test-clients/src/main/java/com/hedera/services/yahcli/commands/contract/00-running-yahcli-to-get-contract-info.md

```diff
@@ -0,0 +1,194 @@
+
+## The signed state "file"
+
+The "signed state file" is not actually a file, it's a directory structure
+containing a merkle tree (in `SignedState.swh`) and peer files/directories
+which are maps of the file store ("storage") and per-smart-contract key-value
+store (these latter two things are "virtual maps").
+
+On disk the signed state file is ~1.8GB (as of 2023-03).  When deserialized
+the entire merkle tree is materialized in memory, as
+well as the indexes to the two virtual maps.  The values of the virtual maps
+are read from disk when accessed.
+
+## Contracts
+
+In the signed state, a contract's bytecode can be found from an account
+id by looking in the file store using a (virtual blob) key which is a
+pair of the account id and `Type.CONTRACT_BYTECODE`.  The account id
+must be of a smart contract account.
+
+## Subcommands of `contract`
+
+The subcommands of the `contract` command are:
+
+* `yahcli contract dumprawcontracts` - dumps the bytecodes for all contracts in the signed state
+    - optionally eliminate duplicates
+* `yahcli contract decompile` - disassembles a contract, given the runtime<sup>‡</sup> bytecode
+    - optionally identifies common code sequences, such as the method selector branch tree ("vtable")
+* `yahcli contract selector`
+    - does selector → method signature lookups on the web
+* (that's it for now)
+
+
+## `dumprawcontracts` subcommand
+
+### get contract bytecodes out of signed state file
+- (signed state file is actually in a directory tree of all signed state information)
+
+```bash
+./yahcli contract dumprawcontracts --with-ids --prefix=">" \
+   -f ~/Downloads/contract_state_hip_583_analysis/SignedState.swh | \
+   grep -E '^>' | cut -f 2,3 > raw-contract-bytecodes.txt
+```
+#### output is one contract/line plus a bunch of cruft to ignore
+- contract lines are preceded by prefix (here `>`)
+    - use `grep -E '^>'` to get _only_ contract lines
+- each contract line has tab-separated fields
+    - first field is prefix (here '>') (if `--prefix` argument given)
+        - above command uses `cut` to remove prefix from line
+    - second field is hex-encoded contract
+    - third field is contract id (decimal) (if `--with-ids`) argument given
+
+- 2023-03-16: found 3396 contracts registered with accounts, 3340 with bytecode (3 with 0-length bytecode! - precompiles?)
+
+Add `--unique` to get only _unique_ contracts (by bytecode), and now, if `--with-ids` argument
+present, each contract line has an additional tab-separated field:
+- fourth field is comma-separated list of all contract ids with this same bytecode
+
+```bash
+./yahcli contract dumprawcontracts --unique --with-ids --prefix=">" \
+   -f ~/Downloads/contract_state_hip_583_analysis/SignedState.swh | \
+   grep -E '^>' | cut -f 2,4 > raw-unique-contract-bytecodes.txt
+```
+
+- 2023-03-16: found 768 unique contracts
+
+### find only solidity contracts
+
+- consider that the solidity prelude is
```

Comment:
Expand on this. Not very common knowledge so let readers know most solidity smart contract bytecode starts with and maps to this and what it means whiles you're at it